packages/blockchain-wallet-v4-frontend/src/modals/Swap/EnterAmount/Checkout/validation.tsx

```diff
@@ -30,34 +30,53 @@ export const getMaxMin = (
         convertBaseToStandard('FIAT', limits.maxPossibleOrder)
       )
       const cryptoMax = new BigNumber(
-        convertStandardToBase(BASE.coin, fiatMax.dividedBy(rate.last))
+        convertStandardToBase(BASE.coin, fiatMax.dividedBy(baseRate.last))
       )
       const userMax = Number(payment ? payment.effectiveBalance : BASE.balance)
       return convertBaseToStandard(
         BASE.coin,
         Math.min(userMax, cryptoMax.toNumber())
       )
     case 'min':
-      const fiatMin = convertBaseToStandard('FIAT', limits.minOrder)
+      // AOTW = As Of This Writing
+      // ℹ️
+      // Let's say you are swapping ETH -> BTC
+      // Let's say the BTC fee is 0.0004517 BTC a.k.a 7 USD (AOTW)
+      // Let's say the minimum you can swap is 3.5 USD
+      // Since 3.5 USD < 7 USD, you would receive -3.5 USD
+      // In that case we say your minimum is the BTC fee + your minimum
+      // 7 USD + 3.5 USD = 10.5 USD
+      // ℹ️
+
+      // ℹ️
+      // In that case you would pay a huge BTC fee, but still end up with
+      // at least the minimum you are allowed swap (send) (3.5 USD)
+      // ℹ️
 
-      // convert limits endpoint min (fiat) to base crypto
-      // using ticker rate
-      const baseMin = new BigNumber(fiatMin).dividedBy(rate.last).toNumber()
+      // /trades/limits is returned in FIAT (3.5 USD)
+      const fiatMin = convertBaseToStandard('FIAT', limits.minOrder)
+      // convert 3.5 USD to BASE coin (0.01123091 ETH)
+      const baseMin = new BigNumber(fiatMin).dividedBy(baseRate.last).toNumber()
 
-      // calculate the network fee of the incoming tx
-      // in base currency using baseMin
-      const counterRate = new BigNumber(1).dividedBy(quote.rate)
+      // calculate the BTC -> ETH rate
+      // 1 BTC = 39.12444194 ETH
+      const exRate = new BigNumber(1).dividedBy(quote.rate)
+      // BTC fee is 0.0004517 BTC a.k.a 4517 satoshi
       const standardCounterFee = convertBaseToStandard(
         COUNTER.coin,
         quote.quote.networkFee
       )
-      const staticFee = convertBaseToStandard(BASE.coin, quote.quote.staticFee)
-      const counterMin = counterRate
-        .times(standardCounterFee)
-        .plus(staticFee)
-        .toNumber()
+      // 4517 satoshi is 0.017672510 ETH is 7 USD (AOTW)
+      const counterFeeInBase = exRate.times(standardCounterFee).toNumber()
 
-      return Math.max(counterMin, baseMin).toPrecision(CRYPTO_DECIMALS)
+      // .017672510 ETH (7 USD) is greater than .01123091 ETH (3.5 USD)
+      if (counterFeeInBase > baseMin) {
```

Comment:
I don't think we need this condition, a better UX would be to always return counterFeeInBase + baseMin, as this wouldn't fix the case where baseMin = 1.0 and counterFeein base = 0.99 for example